### PR TITLE
Wizard recipe: Antlr4CppRuntime-v4.9.3

### PR DESCRIPTION
**Pull request by azreika**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: Antlr4CppRuntime
* Version: v4.9.3

@staticfloat please review and merge.

### CODE REVIEW

File: A/Antlr4CppRuntime/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/antlr/antlr4/archive/refs/tags/$(version).tar.gz", "efe4057d75ab48145d4683100fec7f77d7f87fa258707330cadd1f8e6f7eecae")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/antlr4-4.9.3/runtime/Cpp
+cmake . -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
+make
```
Comment (giordano) [inline A/Antlr4CppRuntime/build_tarballs.jl:17]: Make it more future-proof by removing the reference to the version, build in a subdirectory and enable parallel build
```suggestion
cd $WORKSPACE/srcdir/antlr4*/runtime/Cpp
mkdir build && cd build
cmake .. -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
make -j${nproc}
```

File: A/Antlr4CppRuntime/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
+]
```
Comment (giordano) [inline A/Antlr4CppRuntime/build_tarballs.jl:36]: Let's try to build for all platforms:
```suggestion
platforms = supported_platforms()
```

File: A/Antlr4CppRuntime/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+mkdir build && cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
```
Comment (giordano) [inline A/Antlr4CppRuntime/build_tarballs.jl:25]: See the log:
```
┌ Warning: /agent/_work/1/s/A/Antlr4CppRuntime/build/i686-linux-gnu/73uOJtPb/i686-linux-gnu-libgfortran3-cxx11/destdir/lib/libantlr4-runtime.so.4.9.3 contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/FL6q3/src/auditor/compiler_abi.jl:247
```
```suggestion
platforms = expand_cxxstring_abis(platforms)
```

**Comment by giordano** [comment]
I believe that to make this work on Windows you need first to build https://github.com/nemtrif/utfcpp as a separate package and use it as dependency here.

**Comment by giordano** [comment]
Ok, it was sufficient to forward the CMake toolchain to the local utf8cpp build.  This is looking good now, thanks for your contribution!

**Review by NHDaly — COMMENTED** [review]
🎉 !

**Pull request merged**